### ISSUE TITLE
Does not Sort in One Go

### ISSUE THREAD

**Issue by FlorianWendelborn** [infctr/eslint-plugin-typescript-sort-keys#26]
In very large and nested type definitions, I’ve noticed that this plugin needs up to 5 runs in order to fix all reported sorting issues. First time it finds ~3000, then ~500, then ~50, then 1, then 0.

**Comment by infctr** [comment]
I'd happily accept a PR addressing the issue

**Comment by jamiehaywood** [comment]
@FlorianWendelborn did you start on a PR for fixing this? I'm experiencing the same problem with linting OpenAPI generated type files.

**Comment by adityamatt** [comment]
I have the same problem, is anyone working on it ?

**Comment by FlorianWendelborn** [comment]
I’m not working on it as I no longer need this particular solution. It’s still a good improvement though unless it’s already fixed

**Comment by zachbryant** [comment]
I've started working on this. Too much of a pet peeve to introduce to my repo, everyone will hate me if they have to save 5 times 😅

So far I've added a test with ~400 entries to be sorted. Will be a good baseline to go from

**Comment by zachbryant** [comment]
@adityamatt yep, still planning to create a PR for my changes. The scale is absolutely massive but I'm in the stages of fixing tests to match the new errors

**Comment by zachbryant** [comment]
Just got all of the tests passing. I still want to do some things like testing in supported Node environments and fleshing out test cases, but it's finally come together

**Comment by adityamatt** [comment]
Happy to hear that!!

**Comment by eltonio450** [comment]
hi there :) 

what is the current status of this issue :) ? What is the recommended solution ?

Thank you!
